.github/workflows/build-actions.yml

```diff
@@ -0,0 +1,26 @@
+name: Build-actions
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
I expect something like:
```yaml
on:
  push: 'main'  # triggers the action on push to 'main'

  pull_request: # <---- triggers on PR  
    branches:
      - '*'  # <--- All branches
```

So, what we have:
1. Trigger when pushed to `main`. For example: when you click `merge PR`. So you always know whether your main branch has any issues
2. Trigger when PR opened. So, you can verify whether your PR has no issues